framework/libs/ktest/src/runner.rs

```diff
@@ -110,35 +120,41 @@ where
                     continue;
                 }
             }
-            print!(
-                "test {}::{} ...",
-                test.info().module_path,
-                test.info().fn_name
-            );
+
             debug_assert_eq!(test.info().package, crate_name);
-            match test.run(catch_unwind) {
-                Ok(()) => {
-                    print!(" {}\n", "ok".green());
-                    passed += 1;
-                }
-                Err(e) => {
-                    print!(" {}\n", "FAILED".red());
-                    failed_tests.push((test.clone(), e.clone()));
-                }
-            }
+
+            spawn_ktest(deps, test);
         }
     }
+
+    // Wait for all spawned tests.
+    while FINISHED.load(Ordering::Relaxed) < crate_.nr_tot_tests() {
+        (deps.yield_fn)()
```

Comment:
This is inefficient.

How about re-defining two methods of `KtestDependencies` as follows?

```rust
    /// Spawn a new task, returning the task ID.
    pub spawn_fn: fn(fn() -> ()) -> u32,
    /// Join the task of a given ID.
    pub join_fn: fn(u32),
```

Reply:
Not planned. We haven't got a `join()` method in the frame yet. This is indeed inefficient for FIFO scheduling...